src/servers/main.cc

```diff
@@ -469,21 +600,37 @@ Parse(nvidia::inferenceserver::InferenceServer* server, int argc, char** argv)
     LOG_ERROR << "At least one of the following options must be true: "
               << "--allow-http, --allow-grpc";
     return false;
-  } else if (allow_http && allow_grpc && (http_port == grpc_port)) {
+  } else if (
+      allow_http && allow_grpc &&
+      ((http_port == grpc_port) && ((http_port != -1) && (grpc_port != -1)))) {
```

Comment:
These checks are not really sufficient anymore. Based on the error messages is it clear what actually needs to be checked. If not we can discuss...

Reply:
Moved these tests and added more to CheckPortCollision()